core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -69,9 +70,11 @@ protected ResourcePool(int maxCapacity, ConcurrentLinkedQueue<T> resources) {
    * the pool.
    *
    * @return a resource taken from the pool
+   * @throws IOException if an interruption occurs while acquiring resources
    */
-  public T acquire() {
-    return acquire(null, null);
+  @Override
+  public T acquire() throws IOException {
+    return acquire(WAIT_INDEFINITELY, null);
```

Comment:
I would prefer some really large value here instead of introducing "-1" which just adds complexity unnecessarily. You can keep it consistent with what we are doing in DynamicResourcePool. (simpler is better).

`acquire(100, TimeUnit.DAYS) `